src/lib/Lower.hs

```diff
@@ -587,7 +139,7 @@ traverseTabCon maybeDest tabTy elems = do
     Just  d -> return d
     Nothing -> emitOp $ AllocDest tabTy'
   Abs bord ufoBlock <- buildAbs noHint IdxRepTy \ord -> do
-    buildBlockSimplified $ unsafeFromOrdinal (sink ixTy') $ Var $ sink ord
+    buildBlock $ unsafeFromOrdinal (sink ixTy') $ Var $ sink ord
```

Comment:
How come this works now? Can we handle method projections in Imp?

Reply:
It's handled in Builder, in `applyIxMethod`. I wanted to avoid having Lower produce specialization requests, so I didn't want it to call back into simplification.